### PR TITLE
Optimize if statements that pass through one side

### PR DESCRIPTION
**Pull request by oflatt**

### CODE REVIEW

File: src/rvsdg/to_cfg.rs [hunk 1]
```diff
@@ -461,8 +481,18 @@ impl<'a> RvsdgToCfg<'a> {
...
-                    branch_blocks
-                        .push(self.sequence_results(&[outputs_for_branch, output_assigned]));
+                    if first_passthrough == Some(i) {
+                        // Optimization: if this is the passthrough block, run it first before all other branches
+                        before_if = self.sequence_results(&[
+                            before_if,
+                            outputs_for_branch,
+                            output_assigned,
+                        ]);
+                        branch_blocks.push(self.sequence_results(&[]));
```
Comment (ezrosent) [inline src/rvsdg/to_cfg.rs:491]: So, to clarify we are still creating a block for the passthrough case, but now `optimize_direct_jumps` is going to clean that block up?

**Review by ezrosent — APPROVED** [review]
Looks good! Diff is mostly snapshots. Just one clarification question.

**Review by Alex-Fischman — APPROVED** [review]
LGTM

File: src/rvsdg/to_cfg.rs (hunk 1 above)
Comment (oflatt) [reply src/rvsdg/to_cfg.rs:491]: yes!

**Pull request merged**